### ISSUE TITLE
当我绘制一条跨国家的区域线时,地图层级放大到13级以上,有地图拖拽卡顿的问题v5.2.0

### ISSUE THREAD

**Issue by kingcj1** [openlayers/openlayers#9434]

**Comment by fredj** [comment]
Hi @kingcj1 , please describe the issue in plain English.

**Comment by kingcj1** [comment]
> 您好@ kingcj1，请用简单的英语描述问题。

When I draw a regional line across countries, the map level is enlarged to level 13 or above, and there is a problem with the map dragging the card v5.2.0

> Hi @kingcj1 , please describe the issue in plain English.

When I draw a regional line across countries, the map level is enlarged to level 13 or above, and there is a problem with the map dragging the card v5.2.0

**Comment by kingcj1** [comment]
When the map zoom level is 13 or above, I draw a line cross countries ,  the map operation becomes not smoothy, the version I used is V5.2.0

**Comment by xxlovewei** [comment]
> When the map zoom level is 13 or above, I draw a line cross countries , the map operation becomes not smoothy, the version I used is V5.2.0

Did you draw a normal line or a line with styles? I met the same problem,because my line need calculate more and more times styles when the map zoom in 13 level or above.

**Comment by xxlovewei** [comment]
I've skipped the problem in one way，if your line is the same problem ,I can tell you how to do.

**Comment by kingcj1** [comment]
Perhaps, we met the same problem, so could you tell me how to solve it in detail, such as code about it, thanks

> I've skipped the problem in one way，if your line is the same problem ,I can tell you how to do.

**Comment by xxlovewei** [comment]
> Perhaps, we met the same problem, so could you tell me how to solve it in detail, such as code about it, thanks
> 
> > I've skipped the problem in one way，if your line is the same problem ,I can tell you how to do.


before I set the style of a feature, I first intercept it with a LineString of the screen bound, then I get a new Feature(MultiLineString) and then only set the style of the visible part in screen.

**Comment by xxlovewei** [comment]
Hi @fredj  ,does openlayers have  better solution?
looking forward to your reply~

**Comment by mrpan** [comment]
you  may need to put example to describe the issue. like codepen.io

**Comment by kingcj1** [comment]
The problem has been solved to make vectorlayer({rendermode:image})
